src/main/java/org/jooq/lambda/Iterators.java

```diff
@@ -0,0 +1,116 @@
+package org.jooq.lambda;
+
+import java.util.Iterator;
+import java.util.function.Predicate;
+
+/**
+ * Created by billoneil on 7/26/17.
+ */
+public class Iterators {
+    private enum PredicateCheckState { NOT_TESTED, MORE, ONE_MORE, NO_MORE }
+
+    static <E> PeekingIterator<E> peeking(Iterator<E> iterator) {
+        return new PeekingIteratorImpl<E>(iterator);
+    }
+
+    static <E> Iterator<E> takeWhileIterator(Iterator<E> iterator, Predicate<E> predicate) {
+        return takeWhileIterator(peeking(iterator), predicate);
+    }
+
+    static <E> Iterator<E> takeWhileIterator(PeekingIterator<E> iterator, Predicate<E> predicate) {
```

Comment:
I was trying to use the `Seq.take*` but it closed the stream after the predicate matched and we need it to stay open. I will look into a better way to refactor this.